diff --git a/contrib/torch_memory_saver/python/torch_memory_saver/entrypoint.py b/contrib/torch_memory_saver/python/torch_memory_saver/entrypoint.py
index b14c561..da6c412 100644
--- a/contrib/torch_memory_saver/python/torch_memory_saver/entrypoint.py
+++ b/contrib/torch_memory_saver/python/torch_memory_saver/entrypoint.py
@@ -159,6 +159,7 @@ class _TorchMemorySaverImpl:
             # only be released after the memory region is resumed and empty_cache() is invoked.
             torch_npu._C._npu_releasePool(torch.npu.current_device(), pool.id)
             del pool
+            torch.npu.empty_cache()
         finally:
             self._binary_wrapper.cdll.tms_set_interesting_region(True)
 
